tests/functional/test_install.py

```diff
@@ -1769,6 +1769,26 @@ def test_ignore_yanked_file(script, data):
     assert 'Successfully installed simple-2.0\n' in result.stdout, str(result)
 
 
+def test_index_url_argument(script, data):
+    """
+    Test the behaviour of --index-url arguments
+    """
+
+    # Check if the relevant warning is generated for an invalid index url
+    result = script.pip('install', '--index-url', '--user',
+                        TestPyPI.simple_url, 'simple',
+                        expect_error=True)
+
+    assert 'WARNING: index-url --user is invalid.' in \
+           result.stderr, str(result)
+
+    # Check if the package is successfully installed for a valid index url
+    result = script.pip('install', '--no-deps', '--index-url',
+                        TestPyPI.simple_url, 'simple')
+
+    assert 'Successfully installed simple' in result.stdout, str(result)
+
```

Comment:
I would split this into two separate tests, one for each assert.

Reply:
Thanks @uranusjr , I have split it into two separate tests, please take a look.

Reply:
The comments seem unnecessary to me. The code is quite self-explainatory, and the docstring describes the test well.

Reply:
Hi @uranusjr 

You are correct that the comments were just repeating the docstring, so I have removed them. Please take a look.